File: grider/print_star_based_on_num/print_star_based_on_num.go
package print_star_based_on_num

import (
	"strconv"
)

// 1 * 3 4 * 6 * 8 9 *
// 11 * 13 14 * 16 * 18 19 *
func print(num int) string {
	str := ""

	for i := 1; i <= num; i++ {
		space := " "
		if i == num {
			space = ""
		}

		if i%5 == 2 ||
			i%5 == 0 {

			str = str + "*" + space
			continue
		}

		str = str + strconv.Itoa(i) + space
	}

	return str
}

File: grider/print_star_based_on_num/print_star_based_on_num.py
def printstarbasedonnum(num):
    result = ""
    space = " "
    for i in range(1, num + 1, 1):
        if i == num:
            space = ""

        if i % 5 == 2 or i % 5 == 0:
            result = result + "*" + space
            continue

        result = result + str(i) + space

    return result